obj_factory.h

```diff
@@ -102,6 +102,42 @@ class ObjFactory {
         return vec[idx.val].get();
     }
 
+    bool if_exist(const std::string& uri) {
```

Comment:
I think is usefull when you create your own container to have this method, because we manipulate map inside.
By the way, good catch for `return (map.find(uri) != map.end());` :+1:

Reply:
minor-minor (and that's just an opinion): `exists` seems like a better naming to me.